From f2ca8cd56dd0d5f5e19972764eb8239193db1d67 Mon Sep 17 00:00:00 2001
From: David Redondo <qt@david-redondo.de>
Date: Wed, 7 Oct 2026 19:47:52 +0200
Subject: [PATCH] QDesktopUnixServices: Don't insert empty options when calling
 email portal

The portal validates the options if present. An empty address will be
rejected and the call fails.

Change-Id: I453bb7e476a3d001252f4406ed8dfe469f84946b
Reviewed-by: Liang Qi <liang.qi@qt.io>
(cherry picked from commit 1f9849f7800743b64b5cb00b6bca85a936863a89)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
Origin: upstream, https://invent.kde.org/qt/qt/qtbase/-/commit/f2ca8cd56dd0d5f5e19972764eb8239193db1d67
Forwarded: not-needed
Last-Update: 2026-10-29
---
 src/gui/platform/unix/qdesktopunixservices.cpp | 12 ++++++++----
 1 file changed, 8 insertions(+), 4 deletions(-)

diff --git a/src/gui/platform/unix/qdesktopunixservices.cpp b/src/gui/platform/unix/qdesktopunixservices.cpp
index aa330f226ec..6a0e9ef5e4b 100644
--- a/src/gui/platform/unix/qdesktopunixservices.cpp
+++ b/src/gui/platform/unix/qdesktopunixservices.cpp
@@ -260,9 +260,12 @@ static inline QDBusMessage xdgDesktopPortalSendEmail(const QUrl &url, const QStr
 
     QUrlQuery urlQuery(url);
     QVariantMap options;
-    options.insert("address"_L1, url.path());
-    options.insert("subject"_L1, urlQuery.queryItemValue("subject"_L1));
-    options.insert("body"_L1, urlQuery.queryItemValue("body"_L1));
+    if (const auto address = url.path(); !address.isEmpty())
+        options.insert("address"_L1, address);
+    if (const auto subject = urlQuery.queryItemValue("subject"_L1); !subject.isEmpty())
+        options.insert("subject"_L1, subject);
+    if (const auto body = urlQuery.queryItemValue("body"_L1); !body.isEmpty())
+        options.insert("body"_L1, body);
 
     // O_PATH seems to be present since Linux 2.6.39, which is not case of RHEL 6
 #ifdef O_PATH
@@ -278,7 +281,8 @@ static inline QDBusMessage xdgDesktopPortalSendEmail(const QUrl &url, const QStr
         }
     }
 
-    options.insert("attachment_fds"_L1, QVariant::fromValue(attachments));
+    if (attachments.size() > 0)
+        options.insert("attachment_fds"_L1, QVariant::fromValue(attachments));
 #endif
 
     if (!xdgActivationToken.isEmpty()) {
